File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.js
function solution(n, m, section) {
    const set = new Set(section);
    let answer = 0;
    
    for (let left of section) {
        if (!set.has(left)) continue;
        
        for (let target = left; target < left + m; target++) {
            if (set.has(target)) set.delete(target);
        }
        
        answer++;
        
        if (set.size < 1) break;
    }
    
    return answer;
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section):
    x = section[0] + m - 1
    count = 1
    idx = 0
    
    while x < section[-1]:        
        while section[idx] <= x:
            idx += 1

        x = section[idx] + m - 1
        count += 1
    
    return count